src/app/wallet/wallet.component.html

```diff
@@ -43,94 +43,160 @@ <h4 class="mb-3 text-uppercase">
+                        <div class="transactions__history">
+                            <ng-container *ngFor="let transaction of transactionHistoryPage">
+                                <ng-container *ngIf="shouldShowTransaction(transaction)">
+                                    <div class="transactions__single-element" 
+                                         [class.transactions__error] = "transaction.state !== 'MINED' || transaction.type === 'APPROVE_INVESTMENT'">
+                                        <div class="transactions__left">
+                                            <div class="left__desc">
+                                                <div class="left__transaction-type" 
+                                                    *ngIf="transaction.type === 'INVEST'">
+                                                    Investment <span class="left__conjuctor">to</span>
+                                                </div>
+                                                <div class="left__transaction-type" 
+                                                    *ngIf="transaction.type === 'DEPOSIT'">
+                                                    Deposit
+                                                </div>
+                                                <div class="left__transaction-type" 
+                                                    *ngIf="transaction.type === 'WITHDRAW'">
+                                                    Withdraw
+                                                </div>
+                                                <div class="left__transaction-type" 
+                                                    *ngIf="transaction.type === 'SHARE_PAYOUT'">
+                                                    Dividend <span class="left__conjuctor">from</span>
+                                                </div>
+                                                <div class="left__transaction-type" 
+                                                    *ngIf="transaction.type === 'CANCEL_INVESTMENT'">
+                                                    Cancel investment <span class="left__conjuctor">to</span>
+                                                </div>
+                                                <div class="left__transaction-type" 
+                                                    *ngIf="transaction.type === 'APPROVE_INVESTMENT'">
+                                                    Approve Investment <span class="left__conjuctor">to</span>
+                                                </div>
+                                                {{ transaction.description }}
+                                            </div>
+                                            <div class="left__info">
+                                                <span class="info__state-mined" 
+                                                    *ngIf="transaction.state === 'MINED'">
```

Comment:
This is not meant to be used like that.
transaction.state is an enum, so it should be compared with enums. You can access TransactionState in template by making a varible inside component:

```
transactionState = TransactionState
```

then in template, you use it like:

```
<div *ngIf="transaction.state === transactionState.MINED">...</div>
```

This approach is more secure and less prone to human errors.

Reply:
Oh right, thanks for that info! 😄